File: Assignment3/Class/Boat.java
package vehicles;

public class Boat extends Vehicle{

  private int _capacity;
  private int _occupants;
  private int _roomLeft;

  public Boat(int capacity){
    super();
    _capacity = capacity;
    _occupants = 0;
    _roomLeft = capacity;
  }

  public boolean isYacht(){
    if(_capacity > 50){
       return true;
     }
     return false;
  }

}

File: Assignment5/src/Boat.cpp
#include "Boat.h"

namespace vehicles{
  Boat::Boat(int capacity){
    setCapacity(capacity);
    setOccupants(0);
  }

  bool Boat::isYacht(){
    if(getCapacity() > 50){
      return true;
    }
    return false;
  }
}
